source/GameAction.cpp

```diff
@@ -162,12 +153,11 @@ void GameAction::LoadSingle(const DataNode &child, const string &missionName)
 			specialLogText[child.Token(1)][child.Token(2)] : logText);
 		Dialog::ParseTextNode(child, isSpecial ? 3 : 1, text);
 	}
-	else if(key == "give" && hasValue)
+	else if((key == "give" || key == "take") && child.Size() >= 3 && child.Token(1) == "ship")
 	{
-		if(child.Token(1) == "ship" && child.Size() >= 3)
-			giftShips.emplace_back(GameData::Ships().Get(child.Token(2)), child.Size() >= 4 ? child.Token(3) : "");
-		else
-			child.PrintTrace("Error: Skipping unsupported \"give\" syntax:");
+		ShipManager manager;
+		manager.Load(child);
+		giftShips.emplace_back(manager);
```

Comment:
Either, use `push_back`, or give `ShipManager` a constructor that takes a `DataNode &` and passes it directly to its `Load` method and just do `emplace_back(child)`.
I'm pretty sure what this currently does it create a `ShipManager`, fill it in with its `Load` method, and then construct a new `ShipManager` in the vector, making a copy of the `ShipManager` you pass to `emplace_back` using the immplicit copy constructor.
Or, call `emplace_back` with no arguments and then do `.back().Load(child)`. This might be best because you are going to want the copy constructor for instantiating this action and creating an explicit constructor `ShipManager(const DataNode &node)` might delete all the implicit constructors which you then also have to explicitly set to `default`, I'm not sure.